memote/support/biomass.py

```diff
@@ -176,14 +192,83 @@ def find_direct_metabolites(model, reaction):
         that are taken up to be consumed by the biomass reaction only.
 
     """
-    transport_reactions = set(helpers.find_transport_reactions(model))
-    exchange_reactions = set(model.exchanges)
-    biomass_reactions = set(helpers.find_biomass_reaction(model))
-
-    combined_set = transport_reactions | exchange_reactions | biomass_reactions
-    precursors = find_biomass_precursors(model, reaction)
-
-    return [met for met in precursors if met.reactions.issubset(combined_set)]
+    biomass_rxns = set(helpers.find_biomass_reaction(model))
+    tra_bou_bio_rxns = helpers.find_tra_bou_bio_reactions(model, biomass_rxns)
+    try:
+        precursors = find_biomass_precursors(model, reaction)
+        main_comp = helpers.find_compartment_id_in_model(model, 'c')
+        ext_space = helpers.find_compartment_id_in_model(model, 'e')
+    except KeyError as k:
+        LOGGER.error("Failed to properly identify cytosolic and extracellular "
+                     "compartments.")
+        raise KeyError("The cytosolic and/or extracellular compartments could "
+                       "not be identified. "
+                       "find_compartment_id_in_model: " + str(k))
```

Comment:
If you are raising another error here I don't fully understand why you are catching the `KeyError` in the first place. Or maybe you mean to raise this error with the traceback of the previous error? See `raise_with_traceback` in http://python-future.org/compatible_idioms.html#raising-exceptions

Reply:
Yeah, exactly. I wanted to catch the exception, since finding the proper `c` and `e` compartments was necessary for detecting false positives and realized that models without properly labeled compartments would raise these exceptions, so I wanted to catch them and show the full traceback. I'll change this to `raise_with_traceback`